### PR TITLE
Comment on empty line

### PR DESCRIPTION
**Pull request by polgartom**
This is intended to work almost exactly like in vscode. It doesn't change the existing mechanism; the toggle line comment action only works if empty line or lines are selected, as you can see in the GIF. It's useful if you want to add a line comment for TODO, BUG, etc. I guess I don't need to explain further.

XXXX

### CODE REVIEW

File: src/editors.jai [hunk 1]
```diff
@@ -3155,17 +3155,34 @@ toggle_comment :: (editor: *Editor, buffer: *Buffer, is_fallback := false) {
...
+        
+        editor.cursor_moved = .unimportant;
+    
+        return;
+    }
 
     // Build a replacement string
     b: String_Builder;  // not using temp because the string may be too large
-
+
```
Comment (focus-editor) [inline src/editors.jai:3185]: Whitespace here and on the line earlier above suggests the PR diff wasn't checked before submitting.

File: src/editors.jai [hunk 2]
```diff
@@ -3155,17 +3155,34 @@ toggle_comment :: (editor: *Editor, buffer: *Buffer, is_fallback := false) {
                     break c;
             }
         }
-
-        if is_empty_line { line_starts[it] |= EMPTY_LINE_BIT; continue; }
-        if min_indent < 0 || indent < min_indent then min_indent = indent;
+        
+        if is_empty_line                                      then { line_starts[it] |= EMPTY_LINE_BIT; continue; }
+        if min_indent < 0 || indent < min_indent              then min_indent = indent;
```
Comment (focus-editor) [inline src/editors.jai:3160]: As a rule of thumb when submitting a PR purely stylistic changes should be avoided as they make the code harder to review

File: src/editors.jai [hunk 3]
```diff
@@ -3155,17 +3155,34 @@ toggle_comment :: (editor: *Editor, buffer: *Buffer, is_fallback := false) {
...
         if !add_comment && !begins_with(trimmed_str, comment) then add_comment = true;
     }
 
-    if min_indent < 0 then min_indent = 0;
+    if line_starts.count == 2 && (line_starts[0] & EMPTY_LINE_BIT) && editor.cursors.count == 1 && editor.cursors[0].pos == editor.cursors[0].sel {
```
Comment (focus-editor) [inline src/editors.jai:3164]:
```suggestion
    if editor.cursors.count == 1 && !has_selection(editor.cursors[0]) && (line_starts[0] | EMPTY_LINE_BIT) {
```

File: src/editors.jai [hunk 4]
```diff
@@ -3155,17 +3155,34 @@ toggle_comment :: (editor: *Editor, buffer: *Buffer, is_fallback := false) {
...
+        cursor := *editor.cursors[0];
+        pos := max(cursor.pos, cursor.sel);
+        insert_string_at_offset(buffer, pos, comment);
+        pos += xx comment.count;
+        insert_string_at_offset(buffer, pos, cast(string)" ");
+        pos += 1;
+        
+        cursor.pos = xx pos;
+        cursor.sel = xx pos;
+        cursor.col_wanted = -1;
```
Comment (focus-editor) [inline src/editors.jai:3176]:
```suggestion
        insert_string_at_offset(buffer, cursor.pos, tprint("% ", comment));
        cursor.pos += xx (comment.count + 1);
        cursor.sel = cursor.pos;
        cursor.col_wanted = -1;
```

File: src/editors.jai [hunk 5]
```diff
@@ -3155,17 +3155,34 @@ toggle_comment :: (editor: *Editor, buffer: *Buffer, is_fallback := false) {
...
-
-        if is_empty_line { line_starts[it] |= EMPTY_LINE_BIT; continue; }
-        if min_indent < 0 || indent < min_indent then min_indent = indent;
+        
+        if is_empty_line                                      then { line_starts[it] |= EMPTY_LINE_BIT; continue; }
+        if min_indent < 0 || indent < min_indent              then min_indent = indent;
         if !add_comment && !begins_with(trimmed_str, comment) then add_comment = true;
     }
 
-    if min_indent < 0 then min_indent = 0;
```
Comment (focus-editor) [inline src/editors.jai:3164]: Why was this line removed?

File: src/editors.jai (hunk 2 above)
Comment (polgartom) [reply src/editors.jai:3160]: Yep, I also wanted to stay consistent. It might be harder to review a little bit, but it was easier for me to work with. Besides this procedure already uses this kind of formatting

File: src/editors.jai (hunk 3 above)
Comment (polgartom) [reply src/editors.jai:3164]: Well, I forgot that the has_selection() procedure exists. You used OR instead of AND there. Was that intended?

Comment (focus-editor) [reply src/editors.jai:3164]: No it's a typo

File: src/editors.jai (hunk 5 above)
Comment (polgartom) [reply src/editors.jai:3164]: Unnecessary checking because the add_comment var remains false if min_indent is -1.

**Comment by focus-editor** [comment]
Thanks!
